src/main/java/be/acara/events/service/UserServiceImpl.java

```diff
@@ -61,4 +67,46 @@ public User editUser(Long id, User newUser) {
     public Boolean checkUsername(String username) {
         return userRepository.findByUsername(username) != null;
     }
+
+    @Override
+    public void likeEvent(Long id) {
+        if (!eventRepository.existsById(id)) {
+            throw new EventNotFoundException(String.format("Event with ID %d not found", id));
+        }
+        User user = getCurrentUser();
+        Event event = getEventById(id);
+        event.addUserThatLikesTheEvent(user);
+        eventRepository.saveAndFlush(event);
+    }
+
+    @Override
+    public void dislikeEvent(Long id) {
+        if (!eventRepository.existsById(id)) {
+            throw new EventNotFoundException(String.format("Event with ID %d not found", id));
+        }
+        User user = getCurrentUser();
+        Event event = getEventById(id);
+        event.removeUserThatLikesTheEvent(user);
+        eventRepository.saveAndFlush(event);
+    }
+
+    @Override
+    public boolean doesUserLikeThisEvent(Long id) {
+        User user = getCurrentUser();
+        Event event = getEventById(id);
+        return user.getLikedEvents().contains(event);
+    }
+
+    @Override
+    public User getCurrentUser() {
+        String username = SecurityContextHolder.getContext().getAuthentication().getName();
+        return findByUsername(username);
+    }
+
+    @Override
+    public Event getEventById(Long id) {
+        return eventRepository.findById(id)
+                .orElseThrow(() -> new EventNotFoundException(String.format("Event with ID %d not found", id)));
+    }
```

Comment:
https://www.baeldung.com/circular-dependencies-in-spring

#4.3. Use Setter/Field Injection